README.md

```diff
@@ -2,3 +2,209 @@
+
+- `validators`: is the list of addresses where the validators are listening for incoming monitor requests 
+
+The [_config](cmd/monitor/_config) folder contains some sample config files for the monitor.
+
+### Running the validator
+
+Go to the [validator](cmd/validator) directory inside the [cmd](cmd) package, compile with the following command:
+
+```
+go build
+```  
+
+Then, simply run the generated binary. The validator accepts the following command-line parameters:
+
+- **-config**: path (relative to the project root directory) of the configuration file for the validator (default "/cmd/validator/_config/config_1.yaml")
+
+- **-delay**: time to wait (in seconds) before replying back to the monitor, use for testing (default 0)
+
+The yaml configuration file must have the following parameters in order to provide the validator with the required information to run correctly:
+
+- `id`: unique id of the validator 
+- `address`: address used to listen for incoming requests from the monitor
+- `messages`: list of messages organized with the following structure
+  
+      [height]:
+        heightvoteset:
+          [round]:
+            received_prevote:
+              - type: [PREVOTE | PRECOMMIT]
+                sender: [sender_id]
+                round: [round]
+                value:
+                  data: [value]
+            
+            sent_prevote:
+              - type: [PREVOTE | PRECOMMIT]
+                sender: [sender_id]
+                round: [round]
+                value:
+                  data: [value]
+                
+            
+            received_precommit:
+              - type: [PREVOTE | PRECOMMIT]
+                sender: [sender_id]
+                round: [round]
+                value:
+                  data: [value]
+
+            
+            sent_precommit:
+              - type: [PREVOTE | PRECOMMIT]
+                sender: [sender_id]
+                round: [round]
+                value:
+                  data: [value]
+
+The value in square brackets are values and they are positive integers except for `type` (PREVOTE or PRECOMMIT) and the `data` fields (it can be any integer value, the type can be changed).
+
+The [_config](cmd/validator/_config) folder contains some sample config files for the validator.
+
+
+### Running test scripts
+
+It's possible to run bash scripts (in a Unix environment) in order to run more validator instances and the monitor at the same time and easily test different scenarios.
+A sample bash script is present in the scripts folder and gives a very minimal example of a simple experiment. 
+
+## Acknowledgments
+
+The project is developed as a semester project in collaboration with the [Distributed Computing Lab](https://dcl.epfl.ch/site/) at [EPFL](https://www.epfl.ch/en/).
```

Comment:
You can also add a link to my company, [Informal Systems](informal.systems) if you wish. (We branched out from Interchain Foundation in January.)

Reply:
Done